crates/starknet-server/tests/test_advancing_time.rs

```diff
@@ -552,5 +557,30 @@ mod advancing_time_tests {
             last_block["timestamp"].as_u64(),
             third_increase_time_block["timestamp"].as_u64(),
         );
+
+        // ctrl c
+        send_ctrl_c_signal(&devnet.process).await;
+        std::thread::sleep(std::time::Duration::from_secs(1));
+
+        // load from file and check block number and timestamp
+        let devnet_load = BackgroundDevnet::spawn_with_additional_args(&[
+            "--start-time",
+            past_time.to_string().as_str(),
+            "--dump-path",
+            dump_file.path.as_str(),
+            "--dump-on",
+            "exit",
```

Comment:
In the old implementation, we ALREADY had all the data needed in transactions so there was no need to inform that we wanted to dump anything - data was already there.

Now we need to store this data in a new data structure since not only transactions are re-executed, there is a new approach to handling some new HTTP endpoint events so now the situation has changed. We need to know in advance if the user will want to dump anything to store it.

I decided that if the user doesn't specify dump-on mode in advance dump_events will be just empty to save memory. Of course, we can discuss that.

Reply:
reminder: add new error

Reply:
and update doc

Reply:
Fixed in 51ea8c2a192c37fddadfe0fcfe0ee81742876cf7